logrotate.c

```diff
@@ -2860,6 +2869,36 @@ static int readState(const char *stateFilename)
     return 0;
 }
 
+static int lockState(const char *stateFilename)
+{
+    int lockFd = open(stateFilename, O_RDWR);
```

Comment:
Shouldn't we open the file with `O_CLOEXEC` to avoid leaking the file descriptor to subprocesses?

Reply:
yep, added